sys/clif/clif.c

```diff
@@ -168,22 +177,17 @@ ssize_t clif_add_attr(clif_attr_t *attr, char *buf, size_t maxlen)
     assert(attr);
     assert(attr->key);
 
-    /* if no length given, calculate it */
-    if (!attr->key_len) {
-        attr->key_len = strlen(attr->key);
-    }
-
     /* count attr name size and separator ';' */
     size_t req_space = attr->key_len + 1;
     size_t pos = 0;
-    int quoted = strcmp(attr->key, LF_ATTR_SIZE) ? 1 : 0;
+    int quoted = 0;
+    if(attr->key_len >= 2){
```

Comment:
```suggestion
    if (attr->key_len >= 2){
```

Reply:
Do you mean to compare the size in order to avoid `strcmp`? If so maybe use `LF_ATTR_RES_TYPE_S` instead of the literal.

Reply:
Yes, since the key might not be zero terminated, care must be taken when using strcmp :)